Automattic/pocket-casts-android
# [Project] Up Next Shuffle - Display Unlock Dialog for Shuffle

## Description
- This PR displays the unlock dialog when the user taps on shuffle icon if it's not paid user
- Contex: p1730077568183099-slack-C07T08CTND9
- Figma: KjyDLK1DPPg3snPBMVGWHw-fi-148_3551
- The tracks will be added in #3130
- I asked here about the themes: p1730224482329139/1730077568.183099-slack-C07T08CTND9

Fixes #3128

## Testing Instructions
1. Run the App with free user
2. Play an episode
3. Open Up next
4. ✅  Ensure you see up next shuffle painted as gold color
5. Tap on shuffle icon
6. ✅  Ensure the unlock dialog opens
7. Tap on Not now
8. ✅  Ensure the dialog is dismissed
9. Open the dialog again
10. Tap on Try plus now
11. ✅  Ensure the paywall opens

## Screenshots or Screencast 
[Screen_recording_20241029_163823.webm](https://github.com/user-attachments/assets/f9ee4afa-078d-44b2-b42d-78dc4751c52a)

![image](https://github.com/user-attachments/assets/f0e2fce0-01d3-49c8-920b-cce849ab8444)



## Checklist
 
#### I have tested any UI changes...
